main.go

```diff
@@ -6,80 +6,169 @@ import (
+		case "2":
+			var confnametobook string
+			fmt.Println("Following conferences are available")
+			printConfNames()
+			fmt.Println("Enter the name of the conference you want to book ticket for")
+			fmt.Scan(&confnametobook)
+
+			for k ,conf := range ConfData{
+				if confnametobook==conf.conferenceName && conf.remainingconfTickets > 0 {
+						sharedlib.GreetUser(conf.conferenceName,conf.conferenceTickets,conf.remainingconfTickets)
+						userFirstName , userLastName , userEmail , userTicket := getuserdata()
+						var isuserdatavalid, issue = validateUserData(userFirstName, userLastName, userEmail)
+						if !isuserdatavalid {
+							fmt.Printf("Please submit valid data, issue is in: %v \n", issue)
+							fmt.Print("***************************************************")
+			
+						} else if userTicket > conf.remainingconfTickets {
+							fmt.Printf("Tickets unavailable for the required amount \n")
+						} else {
+							var pointertoconfuser = ConfUser{
+							}
+							conf.remainingconfTickets,pointertoconfuser = makeBooking(userFirstName, userLastName, userEmail, userTicket,conf.remainingconfTickets)
+							conf.Confeventuser = append(conf.Confeventuser, pointertoconfuser) 
+							ConfData[k] =conf
+							fmt.Printf("The name of conference booked is %v \n",conf.conferenceName)
+							fmt.Printf("Number of tickets remaining are %v \n",conf.remainingconfTickets)
+							fmt.Printf("Booking details till now are %v \n",conf.Confeventuser)
+							fmt.Printf("confData: %v\n", ConfData)
+						}
+				}
 			}
 
-		} else {
-			publishBookingdata()
-			break
+		case "3":
+			var confnametoview string
+			fmt.Println("Following conferences are available")
+			printConfNames()
+			fmt.Println("Enter the name of the conference you want to view tickets for")
+			fmt.Scan(&confnametoview)
+
+			for _,conf := range ConfData{
+				if confnametoview == conf.conferenceName {
+					for _,user := range conf.Confeventuser{
+						fmt.Printf("The user %v has booked %v tickets \n",user.firstName +""+user.lastName , user.usertickets)
+					}
+				}
+			}
+			case "4": printConfNames()
+
+
+			default : 
+			fmt.Println("Press a valid option")
+				
+			}
+
+			
 		}
+	}
+/*
+func makeconferencebooking(conf ConferenceEvent, k int){
+	var pointertoconfuser = ConfUser{
+	}
+	conf.remainingconfTickets,pointertoconfuser = makeBooking(userFirstName, userLastName, userEmail, userTicket,conf.remainingconfTickets)
+	conf.Confeventuser = append(conf.Confeventuser, pointertoconfuser) 
+	ConfData[k] =conf
+	fmt.Printf("The name of conference booked is %v \n",conf.conferenceName)
+	fmt.Printf("Number of tickets remaining are %v \n",conf.remainingconfTickets)
+	fmt.Printf("Booking details till now are %v \n",conf.Confeventuser)
+	fmt.Printf("confData: %v\n", ConfData)
+}
+*/
 
+func printConfNames(){
+	for i ,conf := range ConfData{
+		fmt.Printf("Name of the %v conference is %v\n", i+1,conf.conferenceName)
 	}
+}
+func ValidateConfName(confName string) bool {
```

Comment:
This function could be simplified as below:

```go
func ValidateConfName(confName string) bool {
   return slices.ContainsFunc(ConfData, func(event ConferenceEvent) bool {
		return confName == event.conferenceName
	})
}
```

Also, the naming is inconsistent `ValidateConfName` implies a validation is performed, therefore it would return an error. This function just checks that the name exists, perhaps call it `func confNameExists(name string) bool {`

Reply:
I wonder where you saw the hard code of the number of tickets. I guess that was in main branch , that was just to test out, in the new branch , there are no hard codes, hopefully.